engine/runtime/src/test/java/org/enso/interpreter/runtime/ResourceManagerTest.java

```diff
@@ -0,0 +1,62 @@
+package org.enso.interpreter.runtime;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.TruffleObject;
+import com.oracle.truffle.api.library.ExportLibrary;
+import com.oracle.truffle.api.library.ExportMessage;
+import org.enso.interpreter.test.TestBase;
+import org.enso.polyglot.LanguageInfo;
+import org.enso.polyglot.MethodNames;
+import org.junit.Test;
+
+public class ResourceManagerTest extends TestBase {
+
+  public ResourceManagerTest() {}
+
+  @Test
+  public void runFinalizersAtTheEnd() {
+    var obj = new ResourceToGc();
+    var fn = new FnCallback();
+
+    try (var ctx = createDefaultContext()) {
+      ctx.enter();
+      var ensoContext =
+          (EnsoContext)
+              ctx.getBindings(LanguageInfo.ID)
+                  .invokeMember(MethodNames.TopScope.LEAK_CONTEXT)
+                  .asHostObject();
+
+      ensoContext.getResourceManager().register(obj, fn);
+      assertNull("Not invoked yet", fn.args);
+    }
+
+    assertNotNull("Callback invoked when thread closed", fn.args);
+    assertNotNull("Callback invoked with one argument", fn.args.length);
+    assertEquals("Called by our thread", fn.thread, Thread.currentThread());
+    assertEquals("Callback called for SomeObject", obj, fn.args[0]);
```

Comment:
What is `SomeObject` here? It does not seem to refer to anything in this test which is slightly confusing.

Shall we rename this? e.g.
```suggestion
    assertEquals("Callback called for the registered `obj`", obj, fn.args[0]);
```